packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -304,9 +291,63 @@ export class OdspDocumentService implements IDocumentService {
         });
     }
 
+    private async refreshSessionPeriodically(
+        requestWebsocketTokenFromJoinSession: boolean,
+        response: ISocketStorageDiscovery,
+    ) {
+        for (;;) {
+            let refreshSessionDurationSeconds = response.refreshSessionDurationSeconds;
+            if (refreshSessionDurationSeconds !== undefined) {
+                // 30 seconds is buffer time to refresh the session.
+                let retryAfter = (refreshSessionDurationSeconds * 1000) - 30000;
```

Comment:
I think we can remove 30 seconds here, as there is a buffer on SPO side as well (they call it grace period, and I think it's something like 5 minutes). That said, it's worth documenting it here if we rely on it.